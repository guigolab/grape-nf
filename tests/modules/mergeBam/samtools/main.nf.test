nextflow_process {

    name "Test Process mergeBam"
    script "modules/mergeBam/samtools/main.nf"
    process "mergeBam"

    tag "module"

    test("Should merge BAM files using samtools") {

        when {
            params {
                maxMismatches = 4
                maxMultimaps = 10
            }
            process {
                """
                input[0] = Channel.from([
                    [ "sample2", [ "test2", "test3" ], file("${baseDir}/data/test*/*toGenome_sorted.bam"), "bam", "GenomeAlignments", true]
                ])
                """
            }
        }

        then {
            assert process.success
            assert snapshot(process.out).match()
        }

    }

}
