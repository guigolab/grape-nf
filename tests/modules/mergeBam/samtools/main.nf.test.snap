{
    "Should merge BAM files using samtools": {
        "content": [
            {
                "0": [
                    [
                        "sample2",
                        "test2:test3",
                        "sample2_m4_n10_toGenome.bam:md5,9c721443896d97df3f2151ce5397c512",
                        "bam",
                        "GenomeAlignments",
                        true
                    ]
                ]
            }
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "24.04.2"
        },
        "timestamp": "2024-06-17T16:44:46.900499"
    }
}